src/Microsoft.Data.SqlClient/netcore/src/Common/src/Microsoft/Data/ProviderBase/DbConnectionFactory.cs

```diff
@@ -373,6 +378,7 @@ private void PruneConnectionPoolGroups(object state)
                         // otherwise process entry which may move it from active to idle
                         if (entry.Value.Prune())
                         { // may add entries to _poolsToRelease
+                            SqlClientEventSource.Log.ActiveConnectionPoolGroupRequest(false);
```

Comment:
Should we call `ActiveConnectionPoolGroupRequest()` in the else condition? There is another `newConnectionPoolGroups.Add()` called there.

Reply:
The `newConnectionPoolGroups` in the else clause keeps the current connection that its `Prune()` was not successful. So, it means there is no change in the number of active connection pool groups.

Reply:
OK. I see. The `newConnectionPoolGroups` is the subset of the old `_connectionPoolGroups` after we removing the ones to be pruned.